#159987629 User should be able to create articles

emmaadesile opened the pull request:
#### What does this PR do?
Users should be able to create articles

#### Description of Task to be completed?
- users should be able to create articles
- users should be able to add images to their articles
- users should be able to format their text
- users should be able to publish their

#### How should this be manually tested?
- run `git clone <branch url>
- npm install to install app dependencies
- run npm run dev to start the development server
- login to the app
- navigate to `localhost:3000/create-article`
- after writing the article, click publish

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[159987629](https://www.pivotaltracker.com/story/show/159987643)

emmaadesile committed:
ft(create-artilces): users can create articles

- users can apply basic formatting to text

[Deliver 159987629]

emmaadesile committed:
ft(create-article): add support for uploading images

[Deliver 159987629]

emmaadesile committed:
fix merge conflicts from remote branch

emmaadesile committed:
ft(create-articles-test): write tests for create-articles endpoint

- write tests for article action
- write tests for article reducer
- wirte tests for article component

[Delivers 159987629]

emmaadesile committed:
nitpicks(create-article): fix merge conflicts

emmaadesile committed:
ft(create-articles): add validation to article creation

- user cannot create article without title
- user cannot create article without body

[Delivers 159987629]

emmaadesile committed:
ft(create-articles): add more tests

- test util function
- test article action and article reducer

[Delivers 159987629]

emmaadesile committed:
nitpicks(create-articles): add api keys to .env file

[Delivers 159987629]

emmaadesile committed:
nitpicks(create-articles): implement feedback

- rename CREATE_ARTICLE_FAILURE to CREATE_ARTICLE_ERROR
- handle title placeholder focusin and focusout with `titlePlaceholderHandler` method
- handle body placeholder focusin and focusout with `bodyPlaceholderHandler` method
- use Define Plugin in webpack.config.js to define env variables

[Deliver 159987629]

emmaadesile committed:
nitpicks(fix-travis-build): fix failing traivis build

[Delivers 159987629]

emmaadesile committed:
nitpicks(fix-merge-conflicts): fix merge conflicts

- rename article.js to createArticle.js

[Delivers 159987629]

emmaadesile committed:
nitpicks(create-article): update create article initial state

[Delievrs 159987629]

emmaadesile committed:
nitpicks(fix merge-conflicts): resolve merge conflicts from develop

[Delivers 159987629]

emmaadesile committed:
fix user create article feature

emmaadesile committed:
setup app for production

emmaadesile committed:
nitpicks(create-article): fix merge conflicts

- remove duplicate Profile page
- fix merge conflicts

[Delivers 159987629]

d-beloved commented on src/containers/CreateArticle/TextEditor.js:
```diff
@@ -0,0 +1,61 @@
+import { Editor } from "@tinymce/tinymce-react";
+import PropTypes from "prop-types";
+
+const apiKey = "0naso5vy7fw5eu2eqblor00o1c326x0hupermg4cli0rqoqn";
```
please move this into the environment variable

emmaadesile replied:
Done

d-beloved reviewed (changes requested):
A very good job you have done so far.

Please fix the comment I made.

emmaadesile committed:
nitpicks(text-editor): move APIKey to dotenv file

[Delivers 159987629]

emmaadesile committed:
test ecditor to make sure it working correctly

emekafredy reviewed (approved):
LGTM

emmaadesile committed:
ft(create-articles): users can create articles

- user gets success message when an article is successfully created
- user gets an error message if there was an error creating the article
- fix conflicting css rules

[Delivers 159987629]

emmaadesile committed:
Merge branch 'develop' into ft-user-create-articles-159987629

johnkingzy reviewed (approved):
Nice one!
